Content.Server/GameTicking/GameTicker.RoundFlow.cs

```diff
@@ -153,14 +158,36 @@ public async void StartRound(bool force = false)
                 var profile = profiles[player.UserId];
                 if (profile.PreferenceUnavailable == PreferenceUnavailableMode.SpawnAsOverflow)
                 {
-                    assignedJobs.Add(player, OverflowJob);
+                    // Pick a random station
+                    var stations = _stationSystem.StationInfo.Keys.ToList();
+                    _robustRandom.Shuffle(stations);
+
+                    if (stations.Count == 0)
+                    {
+                        assignedJobs.Add(player, (FallbackOverflowJob, StationSystem.StationId.Invalid));
+                        continue;
+                    }
+
+                    foreach (var station in stations)
+                    {
+                        // Pick a random overflow job from that station
+                        var overflows = _stationSystem.StationInfo[station].MapPrototype.OverflowJobs.Clone();
+                        _robustRandom.Shuffle(overflows);
+
+                        // Stations with no overflow slots should simply get skipped over.
+                        if (overflows.Count == 0)
+                            continue;
+
+                        // If the overflow exists, put them in as it.
+                        assignedJobs.Add(player, (overflows[0], stations[0]));
+                    }
                 }
             }
 
             // Spawn everybody in!
             foreach (var (player, job) in assignedJobs)
             {
-                SpawnPlayer(player, profiles[player.UserId], job, false);
+                SpawnPlayer(player, profiles[player.UserId], job.Item2, job.Item1, false);
```

Comment:
might as well just make it return a third item in the tuple since 'job.Item2' being the station ID is a little weird